sys/fork.c

```diff
@@ -58,6 +58,8 @@ int do_fork(void) {
   memcpy(child->p_sigactions, parent->p_sigactions,
          sizeof(child->p_sigactions));
 
```

Comment:
Is it ok that we access `child->p_uspace` / `parent->p_uspace` in https://github.com/cahirwpz/mimiker/blob/925c40d16a16e4465cd7d9c90441fce43478ffbd/sys/fork.c#L48 and `p_uspace` is marked `($) use only from the same process/thread` ?

Reply:
This is why there is distinction between proc_create and proc_add. The process is invisible by other patrs of the system untill proc_add is called, so this is safe.

Reply:
Is it safe to access  `parent->p_waitcv` here: https://github.com/cahirwpz/mimiker/blob/4e01cad7070f03f5571a09e0854a5b5c04d9b798/sys/proc.c#L175 ?

Reply:
Yes, it is locked using global mutex in our implementation.

Reply:
It's always safe to call `cv_broadcast` or `cv_signal` even in interrupt context.  You must always call `cv_wait` with same mutex.

Reply:
In this case, it's safe as long as you're sure nobody will destroy `parent`.